src/native/libs/System.Native/pal_uid.c

```diff
@@ -276,9 +279,15 @@ char* SystemNative_GetGroupName(uint32_t gid)
         bufferLength = tmpBufferLength;
     }
 #else
-    (void)gid;
-    errno = ENOTSUP;
-    return NULL;
+    // Platforms like Android API level < 24 do not have getgrgid_r available
+    // Android API level >= 24 do have getgrgid_r available but call getgrgid internally anyway
+    result = getgrgid(gid);
+    if (result == NULL)
+    {
+        return NULL;
+    }
+    name = strdup(result->gr_name);
+    return name;
```

Comment:
Added fallback code for when `getgrgid_r` is not found, which calls `getgrgid` instead.